CHANGELOG.md

```diff
@@ -4,18 +4,26 @@
 
 **Breaking Changes**
 
-- Support for `alertingRules[].summary` was dropped in favor of `alertingRules[].description`. Although backwards-compatible this is silently breaking: Defined summaries will no longer be propagated.
+> Support for `alertingRules[].summary` was dropped in favor of `alertingRules[].description`. Although backwards-compatible this is silently breaking: Defined summaries will no longer be propagated.
 
 ## 3.0.0
 
 **Breaking Changes**
 
-- Support for `livenessProbe.delay` and `readinessProbe.delay` was dropped in favor of `startupProbe.periodSeconds` and `startupProbe.failAfterCount`
+> Support for `livenessProbe.delay` and `readinessProbe.delay` was dropped in favor of `startupProbe.periodSeconds` and `startupProbe.failAfterCount`
 
 - Add `livenessProbe.failAfterCount` and `readinessProbe.failAfterCount` to control how many must fail before corrective action is taken
 
 ## 2.2.1
 
+**Breaking changes**
+
+> :warning: **WARNING**: The first time *2.2.1 or any version after* is deployed when *any version before 2.2.1* is running this causes a **nonzero downtime deployment** with a short 503 service disruption due to a [bug in the ingress controller](https://github.com/kubernetes/ingress-nginx/issues/6962).
```

Comment:
Can't we simply remove the name from the port, to prevent any non-zero downtime deployment? I see little advantage of using the name when it causes such a headache coordinating such a version upgrade. And I'm sure someone will NOT read this disclaimer.

Reply:
The name is required once you want to [expose multiple ports](https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services):

> When using multiple ports for a Service, you must give all of your ports names so that these are unambiguous.

So for one this would require making the port name conditional on if you're using multiple ports which maybe could be confusing in the future?

But more importantly although that would reduce the blast radius *right now*, it also would cause the same behavior repeatedly and not only depending on the version of the chart but also on the local values: every time you deploy when the deployed service does not yet use multiple ports and your service to deploy will use at least two, it will cause the same issue again. That sounds even worse for me both in terms of usability and if/how to get the warning across.

Reply:
Okay. Like discussed, we are coordinating the update in all services.